### PR TITLE
Infer multi-threads API Demo and UT

### PR DESCRIPTION
**Pull request by tensor-tang**
- [x] Demo
- [x] UT
  - [x] word2vec
  - [x] image_classification

### CODE REVIEW

**Comment by tensor-tang** [comment]
[Step 1/1] make[2]: *** [paddle/fluid/inference/analysis/test_dfg_graphviz_draw_pass] Error 1

https://paddleci.ngrok.io/viewLog.html?tab=buildLog&logTab=tree&filter=debug&expand=all&buildId=58&_focus=5046

File: paddle/contrib/inference/demo/simple_on_word2vec.cc [hunk 1]
```diff
@@ -61,10 +61,60 @@ void Main(bool use_gpu) {
     for (size_t i = 0; i < std::min(5UL, num_elements); i++) {
       LOG(INFO) << static_cast<float*>(outputs.front().data.data)[i];
     }
+    // TODO(Superjomn): this is should be free automatically
+    free(outputs[0].data.data);
+  }
+}
+
+void MainThreads(int num_threads) {
```
[the remark this answers is not in the document]
Comment (tensor-tang) [reply paddle/contrib/inference/demo/simple_on_word2vec.cc]: Sure~

File: paddle/contrib/inference/demo/simple_on_word2vec.cc [hunk 2]
```diff
@@ -61,10 +61,60 @@ void Main(bool use_gpu) {
...
+  config.model_dir = FLAGS_dirname + "word2vec.inference.model";
+  config.use_gpu = false;
+  auto main_predictor =
+      CreatePaddlePredictor<NativeConfig, PaddleEngineKind::kNative>(config);
+
+  std::vector<std::thread> threads;
+  for (int tid = 0; tid < num_threads; ++tid) {
+    threads.emplace_back([&, tid]() {
+      // 1. clone a predictor which shares the same parameters
+      auto predictor = main_predictor->Clone();
```
[the remark this answers is not in the document]
Comment (tensor-tang) [reply paddle/contrib/inference/demo/simple_on_word2vec.cc]: I am afraid it's better to be inside thread.

**Review by Superjomn — APPROVED** [review]
LGTM

**Pull request merged**